src/panel/direction/MobileRoadMapPreview.jsx

```diff
@@ -2,62 +2,79 @@ import React from 'react';
     onClose: PropTypes.func.isRequired,
-  }
+  };
 
-  state = {
-    currentStep: 0,
-  }
+  state = { };
 
   componentDidMount() {
-    this.zoomToCurrentStep();
-    fire('move_mobile_geolocation_button', 65);
+    this.steps = document.querySelector('.mobile-roadmap-preview-steps');
+    this.stepList = document.querySelectorAll('.itinerary_mobile_step');
+    this.currentStep = 0;
+    fire('move_mobile_bottom_ui', 105);
+    this.highlightCurrentStep();
   }
 
-  zoomToCurrentStep = () => {
-    fire('zoom_step', this.props.steps[this.state.currentStep]);
-  }
+  highlightCurrentStep = () => {
+    fire('zoom_step', this.props.steps[this.currentStep]);
+    this.stepList.forEach((item, index) => {
 
-  previous = () => {
-    this.setState(
-      prevState => ({ currentStep: Math.max(prevState.currentStep - 1, 0) }),
-      this.zoomToCurrentStep
-    );
-  }
+      // Reset the style of all the steps
+      item.classList.remove('active');
+      item.classList.remove('past');
+
+      // Show previous steps as "past"
+      if (index < this.currentStep) {
+        item.classList.add('past');
+      }
+    });
+
+    // Show current step as "active"
+    this.stepList[this.currentStep].classList.add('active');
+  };
 
-  next = () => {
-    this.setState(
-      prevState => ({ currentStep: Math.min(prevState.currentStep + 1, this.props.steps.length) }),
-      this.zoomToCurrentStep
+  scroll = () => {
+
+    const currentStep = Math.floor(
+      // Divide the step container's scrollLeft up to the middle of the screen with the size of a step
+      // to determine which step is present at the middle of the screen
+      (this.steps.scrollLeft + window.innerWidth / 2) / (window.innerWidth - 70 + 12)
     );
-  }
 
-  render() {
-    const currentStep = this.props.steps[this.state.currentStep];
+    // If it has changed, save it and highlight it as the current step
+    if (this.currentStep !== currentStep) {
+      this.currentStep = currentStep;
+      this.highlightCurrentStep();
+    }
+  };
 
-    return <React.Fragment>
-      <div className="itinerary_mobile_step">
-        <CloseButton onClick={this.props.onClose} />
-        <RoadMapStep step={currentStep} />
-      </div>
+  toggleSize = () => {
+    this.steps.classList.toggle('expanded');
+    fire('move_mobile_bottom_ui', this.steps.offsetHeight);
+  };
 
-      <div className="itinerary_mobile_step_buttons">
-        <button
-          className="itinerary_mobile_step_button icon-chevron-left"
-          disabled={this.state.currentStep === 0}
-          onClick={this.previous}
-        />
-        <button
-          className="itinerary_mobile_step_button icon-chevron-right"
-          disabled={this.state.currentStep === this.props.steps.length - 1}
-          onClick={this.next}
-        />
+  render() {
+    return <div className="itinerary_mobile_step_by_step">
+      <div className="mobile-roadmap-preview-close" onClick={this.props.onClose}>
```

Comment:
This could probably use the `<Button>` component, which creates a real `button` element.

Reply:
I know it's consistent with the design, but this back button looks awfully small. Was it discussed?
Anyway, if not it's ok to move on with this size and come back on it later.